Realm/RLMRealm.mm

```diff
@@ -516,8 +557,7 @@ + (instancetype)realmWithConfiguration:(RLMRealmConfiguration *)configuration er
         RLMRealmCreateAccessors(realm.schema);
 
         if (!readOnly) {
-            // initializing the schema started a read transaction, so end it
-            [realm invalidate];
+            REALM_ASSERT(!realm->_realm->is_in_read_transaction());
```

Comment:
[q] Is this unrelated to the rest of the PR? Are adding an assertion instead of allowing this?

Reply:
I tried to make it so that Realms could be opened for dispatch queues other than the current one, and part of that was making it so that update_schema() ends the read transaction it starts rather than this code. Supporting that turned out to be harder than it was worth, but I think it's still worth leaving in the change to update_schema()'s behavior. There's no user-visible difference here.

The reason we don't want newly created Realm instances to have a read transaction is to make code like `while (true) dispatch_async(queue, ^{ [RLMRealm.defaultRealm transactionWithBlock:^{ ... }]);` less bad. Starting a read transaction as part of the RLMRealm initialization means that all the threads waiting for the write lock are pinning a Realm version and the file size will grow rapidly, while with this behavior there's no pinning and it'll stay fairly constant.